[Feat] Enable `vm.Switch` as a selector for `Filter` and `Parameter` (#1286)
mckinsey/vizro: vizro-core/docs/pages/user-guides/filters.md

## Change selector

> If you want to have a different selector for your filter, you can give the `selector` argument of the [`Filter`][vizro.models.Filter] a different selector model. Currently available selectors are [`Checklist`][vizro.models.Checklist], [`Dropdown`][vizro.models.Dropdown], [`RadioItems`][vizro.models.RadioItems], [`RangeSlider`][vizro.models.RangeSlider], [`Slider`][vizro.models.Slider], and [`DatePicker`][vizro.models.DatePicker].

Reviewer: Should we have an example in the docs of a filter on a boolean column? Or just update our features demo and link to that?

As per my comments on https://github.com/McK-Internal/vizro-internal/issues/1992 the docs on filters/selectors in general are pretty brief and you don't really get a good impression of what the different selector types and settings look like at the moment. Doesn't need to be part of this PR, but at some point I think we should try to improve that, even if it's just by saying "look at this demo page to see all the selectors you can use".

Reviewer 2: Yes, for now I would just refer to our dev demo - we have all selectors there and I just updated it in this PR: https://vizro-demo-features.hf.space/selectors

Reviewer: Nice, thanks for doing this!

If you're updating it anyway then should we also update the HF demo to the latest Vizro version so we get the new "select all" functionality in there?

## Change selector

If you want to have a different selector for your filter, you can give the `selector` argument of the [`Filter`][vizro.models.Filter] a different selector model. Currently available selectors are [`Checklist`][vizro.models.Checklist], [`Dropdown`][vizro.models.Dropdown], [`RadioItems`][vizro.models.RadioItems], [`RangeSlider`][vizro.models.RangeSlider], [`Slider`][vizro.models.Slider], [`DatePicker`][vizro.models.DatePicker] and [`Switch`][vizro.models.Switch].

> You can explore and test all available selectors interactively at our [demo page](https://vizro-demo-features.hf.space/selectors).

Reviewer: ```suggestion
You can explore and test all available selectors interactively on our [feature demo dashboard](https://vizro-demo-features.hf.space/selectors).
```
